### ISSUE TITLE
How to implement the plugin within a Moodle plugin

### ISSUE THREAD

**Issue by greg-pedder** [julianna-langston/chartjs-plugin-a11y-legend#181]
Hi there,
I'm writing a plugin in Moodle which uses Chart JS - specficially the Doughnut/Bar chart - however, the charts lack any kind of accessibility features. Looking to use your plugin within Chart JS as the keyboard navigation is exactly what I'm looking for which would help meet our accessibility requirements.
Without having to go down the NPM route to install this, and having tried the Codepen example (https://codepen.io/chart2music/pen/ZEMyLVZ) - which unfortunately doesn't work for me - is there a way to get this running in Moodle?

Currently, trying to get this running using the Codepen example gives me the following console log errors:
Uncaught SyntaxError: Unexpected token 'export' (at chartjs-plugin-a11y-legend@0.0.2:15:1)

**Comment by julianna-langston** [comment]
A couple clarifying questions -
1. Are you seeing the errors in CodePen itself, or are you seeing the errors when you copy the code somewhere else?
2. What operating system and browser combination are you using?
3. Are you referring to the keyboard navigation *within* the chart, or are you looking for keyboard navigation for the chart *legend*? The chartjs-plugin-a11y-legend repo only manages the keyboard navigation of the *legend*, whereas [the chartjs2music](https://github.com/julianna-langston/chartjs2music) repo handles the keyboard navigation for the chart *data*.

All that said, if you are getting a syntax error on the keyword "export", you might need another version. CodePen supports ES6 imports and exports, but if your build environment doesn't, you might want to try linking to this URL:
https://[cdn.jsdelivr.net/npm/chartjs-plugin-a11y-legend](https://cdn.jsdelivr.net/npm/chartjs-plugin-a11y-legend)

If you add that script, it should initialize a global variable named `chartjsPluginA11yLegend`, which you would use to register the Chart.js plugin. ([Chart.js plugin documentation](https://www.chartjs.org/docs/latest/developers/plugins.html)).

**Comment by greg-pedder** [comment]
Hi Julianna - thanks for getting back so quickly.

1) Not in CodePen itself, it was when using the import statement 'import a11yLegend from "https://cdn.skypack.dev/chartjs-plugin-a11y-legend@0.0.2";' in my script. Not sure how familiar you are with how Moodle does things, but it's an AMD module that is getting compiled using Grunt.
2) Running this using Multipass on a Mac - basically, Docker containers set up to run web/db servers via Ubuntu. Browsers used are Chrome/FF/Opera.
3) Ideally it would/could be for both. I've managed to get the legends and tooltips running with their own onclick handlers - the plugin that you have developed will be ideal for adding keyboard navigation ^to^ the legends essentially - I have therefore not used the chartjs2music code as I'm happy to see this working just for the legends to begin with.

I will try your suggested link in the meantime.

Thanks again.

**Comment by greg-pedder** [comment]
Hi Julianna - that link didn't work I'm afraid.
That would be great if you could open the PR when you get a chance.
Thanks again!

**Comment by julianna-langston** [comment]
Just to set expectations, it will probably be another week or 2 until that comes through, but this is on my list.

**Comment by greg-pedder** [comment]
No problems - thanks again for accommodating this request.

**Comment by julianna-langston** [comment]
Had some unexpected free time, so I took a stab at this. I'm now outputting AMD format in v0.2.0. 

[Here is a direct link to a minified, AMD-formatted js file](https://cdn.jsdelivr.net/npm/chartjs-plugin-a11y-legend@0.2.0/dist/plugin.amd.min.js). [Here's the unminified version.](https://cdn.jsdelivr.net/npm/chartjs-plugin-a11y-legend@0.2.0/dist/plugin.amd.js)

I haven't used anything in an AMD format in a long time, so I'm not really sure how to test it... Do you mind giving that file a try, and let me know how it works?

**Comment by greg-pedder** [comment]
That seems to have worked perfectly - thanks again for making those changes.
I'll get some end users to check that all is working as expected, but for now that change seems ideal.

**Comment by julianna-langston** [comment]
Awesome! I'm glad that worked. 

Note, I've opened [an issue in the other plugin](https://github.com/julianna-langston/chartjs2music/issues/242) to support AMD formats, so that it'll be there for you when you need it.
